zebedee/client_test.go

```diff
@@ -607,3 +608,95 @@ func TestClient_PublishedDataEndpoint(t *testing.T) {
 		})
 	})
 }
+
+func TestClient_PublishedIndexEndpoint(t *testing.T) {
+	ctx := context.Background()
+	path := "/publishedindex"
+
+	Convey("given a 200 response", t, func() {
+		documentContent, err := ioutil.ReadFile("./response_mocks/publishedindex.json")
+		So(err, ShouldBeNil)
+		body := httpmocks.NewReadCloserMock(documentContent, nil)
+		response := httpmocks.NewResponseMock(body, http.StatusOK)
+
+		httpClient := newMockHTTPClient(response, nil)
+		zebedeeClient := newZebedeeClient(httpClient)
+
+		Convey("when zebedeeClient.GetPublishedIndex is called", func() {
+			m, err := zebedeeClient.GetPublishedIndex(ctx, nil)
+
+			Convey("then the expected content is returned", func() {
+				So(err, ShouldBeNil)
+				So(m.Count, ShouldEqual, 3)
+				So(m.Items, ShouldNotBeEmpty)
+				So(m.Items, ShouldHaveLength, 3)
+				mItem := m.Items[0]
+				So(mItem.URI, ShouldResemble, "/releases")
+				So(m.Limit, ShouldEqual, 3)
+				So(m.Offset, ShouldEqual, 0)
+				So(m.TotalCount, ShouldEqual, 3)
+			})
+
+			Convey("and client.Do should be called once with the expected parameters", func() {
+				doCalls := httpClient.DoCalls()
+				So(doCalls, ShouldHaveLength, 1)
+				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
+				//p := doCalls[0].Req.FormValue("offset")
+				//So(p, ShouldEqual, path)
+			})
+		})
+
+		Convey("when zebedeeClient.GetPublishedIndex is called with paging params", func() {
+			limit := 9
+			m, err := zebedeeClient.GetPublishedIndex(ctx, &PublishedIndexRequestParams{
+				Limit:  &limit,
+				Offset: 5,
+			})
+
+			Convey("then the expected content is returned", func() {
+				So(err, ShouldBeNil)
+				So(m.Count, ShouldEqual, 3)
+				So(m.Items, ShouldNotBeEmpty)
+				So(m.Items, ShouldHaveLength, 3)
+				mItem := m.Items[0]
+				So(mItem.URI, ShouldResemble, "/releases")
+				So(m.Limit, ShouldEqual, 3)
+				So(m.Offset, ShouldEqual, 0)
+				So(m.TotalCount, ShouldEqual, 3)
+			})
+
+			Convey("and client.Do should be called once with the expected parameters", func() {
+				doCalls := httpClient.DoCalls()
+				So(doCalls, ShouldHaveLength, 1)
+				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
+				req := doCalls[0].Req
+				So(req.FormValue("offset"), ShouldEqual, "5")
+				So(req.FormValue("limit"), ShouldEqual, "9")
+			})
+		})
+	})
+
```

Comment:
Do 404 responses from Zebedee count as errors (in which case no extra test needed), or as empty collections?

Reply:
Zebedee never returns 404 unless the wrong thing was requested

Reply:
No 404 test is needed for the index as it's not specifying a specific resource to return.